doc/holochain_101/src/zome/read_and_write.md

```diff
@@ -1 +1,154 @@
+
+Because the Entry will be permanent in your local source chain, it must first pass validation, in order to be written in the first place. Validation of Entries is core to the distributed data integrity model of Holochain.
+
+Invoking the commit entry API function will not always return "success". The call can fail if the Entry fails to pass validation, or something else goes wrong internally while Holochain is attempting to write the Entry to the local source chain. Like in any code, failure tolerant code is far better, and Zomes should be written to handle possible error cases.
+
+So in general, the process that Holochain follows while trying to write an Entry during a "commit" call is this:
+
+1. Collect data necessary for use in validation (a "validation package")
+2. Check whether the new entry is valid by passing it through the validation callback specified in the Zome for its' given entry type. If this fails, it will exit here and return either a user defined, or system level error message
+3. Creates a new Header for the Entry, and writes it to storage
+4. Writes the Entry itself to storage
+5. Announces over the network module this new Header and Entry, requesting that peers validate and hold a copy of it. This is known internally as "publishing".
+6. Returns the "address" of the new Entry, which is the crytographic hash of the Entry data combined with its entry type string. This address can be used later to retrieve the Entry.
+
+
+### Updating Entries
+
+The act of updating an Entry is technically speaking just the same "committing an Entry", plus storing a bit of metadata indicating that this new entry is an "update" to the entry at an old address which is also given. This has the effect that when an attempt to retrieve an Entry by its address, it will forward the request along to the latest version in a potential series of updates, and return that instead. There is always the option to return the results for an Entry according to whether it's the "initial" version, the "latest", or getting the entire history too.
+
+So to update an Entry, use the address of that entry, and provide the new Entry to replace it. You can use the address of the Entry at any point in its update history as the address to update (as long as it hasn't been marked deleted), and it will still work, technically updating the very latest version of that Entry instead of whatever you pass in.
+
+So in general, the process that Holochain follows while trying to update an Entry during an "update_entry" call is this:
+
+1. Retrieve the very latest version of the Entry at the given address
+2. Collect data necessary for use in validation (a "validation package")
+3. Check whether the new entry is valid by passing it through the validation callback specified in the Zome for its' given entry type. If this fails, it will exit here and return either a user defined, or system level error message
+4. Creates a new Header for the Entry, and writes it to storage
```

Comment:
I think this workflow description is missing a few steps. Whether you want to leave these out for clarity it is up to you but there is also:
- Checks if the validation requirements are met for updating the given entry. The validation callback for the entry is called again with the context that an update is being attempted. This allows the DNA developer to enforce conditions on updating (e.g. only the author of an entry can update it)
- If the above passes it then commits an UpdateEntry system entry logging the fact that this modification took place. (Actually is this true? I could't find anything about this in the core code...)

Reply:
"If the above passes it then commits an UpdateEntry" 
no, there's no such entry type. It is just metadata sitting in the EAV. 

Step 3 for this could be revised to be clearer, but this is the validation step you mention. It is given both the old entry, and the new entry, along with the headers, which it uses to decide if this action should complete successfully or not